source/fuzz/fuzzer_pass_outline_selection_constructs.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2020 Vasyl Teliman
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/fuzzer_pass_outline_selection_constructs.h"
+
+#include "source/fuzz/fuzzer_context.h"
+#include "source/fuzz/instruction_descriptor.h"
+#include "source/fuzz/transformation_outline_selection_construct.h"
+#include "source/fuzz/transformation_split_block.h"
+
+namespace spvtools {
+namespace fuzz {
+
+FuzzerPassOutlineSelectionConstructs::FuzzerPassOutlineSelectionConstructs(
+    opt::IRContext* ir_context, TransformationContext* transformation_context,
+    FuzzerContext* fuzzer_context,
+    protobufs::TransformationSequence* transformations)
+    : FuzzerPass(ir_context, transformation_context, fuzzer_context,
+                 transformations) {}
+
+FuzzerPassOutlineSelectionConstructs::~FuzzerPassOutlineSelectionConstructs() =
+    default;
+
+void FuzzerPassOutlineSelectionConstructs::Apply() {
```

Comment:
Did you look at FuzzerPassOutlineFunction?  The way this looks for a single-entry-single-exit region is kind of similar.  Is there scope for unifying them?

Reply:
That said, it might well be not worth it.  The conditions on the region that you require here are a bit different from the conditions required for outlining into functions.

Reply:
Indeed, conditions are a bit different between the two transformations. I left it as it is.